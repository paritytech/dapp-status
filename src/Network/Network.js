// Copyright 2015-2017 Parity Technologies (UK) Ltd.
// This file is part of Parity.

// Parity is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

// Parity is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.

// You should have received a copy of the GNU General Public License
// along with Parity.  If not, see <http://www.gnu.org/licenses/>.

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { inject, observer } from 'mobx-react';
import Form from 'semantic-ui-react/dist/commonjs/collections/Form';
import Statistic from 'semantic-ui-react/dist/commonjs/views/Statistic';
import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
import formatNumber from 'format-number';

import Field from '../components/Field';
import Section from '../components/Section';
import styles from './Network.css';

const toNiceNumber = formatNumber();

class Network extends Component {
  static propTypes = {
    intl: intlShape
  };

  render() {
    const {
      chainStore: { chain },
      enodeStore: { enode },
      netPeersStore: { netPeers },
      netPortStore: { netPort },
      rpcSettingsStore: { rpcSettings },
      intl: { formatMessage }
    } = this.props;

    return (
      <Section
        title={
          <FormattedMessage
            id="dapp.status.network.title"
            defaultMessage="Network Settings"
          />
        }
      >
        <Form>
          <Field
            label={
              <FormattedMessage
                id="dapp.status.network.chainLabel"
                defaultMessage="Chain"
              />
            }
            readOnly
            value={chain}
          />
          <Form.Group widths={2}>
            <Field
              label={
                <FormattedMessage
                  id="dapp.status.network.rpcEnabled"
                  defaultMessage="RPC Enabled"
                />
              }
              readOnly
              value={
                rpcSettings && rpcSettings.enabled
                  ? formatMessage(messages.rpcEnabledYes)
                  : formatMessage(messages.rpcEnabledNo)
              }
            />
            <Field
              label={
                <FormattedMessage
                  id="dapp.status.network.netPort"
                  defaultMessage="Network Port"
                />
              }
              readOnly
              value={netPort}
            />
          </Form.Group>
          <Form.Group widths={2}>
            <Field
              label={
                <FormattedMessage
                  id="dapp.status.network.rpcInterface"
                  defaultMessage="RPC Interface"
                />
              }
              readOnly
              value={rpcSettings && rpcSettings.interface}
            />
            <Field
              label={
                <FormattedMessage
                  id="dapp.status.network.rpcPort"
                  defaultMessage="RPC Port"
                />
              }
              readOnly
              value={rpcSettings && rpcSettings.port}
            />
          </Form.Group>
          <Field
            label={
              <FormattedMessage
                id="dapp.status.network.enode"
                defaultMessage="Enode"
              />
            }
            readOnly
            showCopyButton
            value={enode}
          />
        </Form>
        <Statistic.Group
          size="tiny"
          widths={1}
          className={styles.smallStatistic}
        >
          <Statistic>
            <Statistic.Value>{`${toNiceNumber(netPeers.active) ||
              0}/${toNiceNumber(netPeers.connected) || 0}/${toNiceNumber(
              netPeers.max
            ) || 0}`}</Statistic.Value>
            <Statistic.Label className={styles.lowerCaseLabel}>
              <FormattedMessage
                id="dapp.status.mining.netPeersLabel"
                defaultMessage="Active/Connected/Max Peers"
              />
            </Statistic.Label>
          </Statistic>
        </Statistic.Group>
      </Section>
    );
  }
}

export default injectIntl(
  inject(
    'chainStore',
    'enodeStore',
    'netPeersStore',
    'netPortStore',
    'rpcSettingsStore'
  )(observer(Network))
);

const messages = {
  rpcEnabledYes: {
    id: 'dapp.status.network.rpcEnabledYes',
    defaultMessage: 'Yes'
  },
  rpcEnabledNo: {
    id: 'dapp.status.network.rpcEnabledNo',
    defaultMessage: 'No'
  }
};
