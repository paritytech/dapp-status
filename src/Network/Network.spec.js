// Copyright 2015-2017 Parity Technologies (UK) Ltd.
// This file is part of Parity.

// Parity is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

// Parity is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.

// You should have received a copy of the GNU General Public License
// along with Parity.  If not, see <http://www.gnu.org/licenses/>.

import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import BigNumber from 'bignumber.js';

import { shallowWithIntl } from '../setupTests';
import { Network } from './Network';

const props = {
  chainStore: { chain: 'Foo' },
  enodeStore: { enode: 'Bar' },
  netPeersStore: {
    netPeers: {
      active: BigNumber(1),
      connected: BigNumber(2),
      max: BigNumber(3)
    }
  },
  netPortStore: { netPort: BigNumber(4) },
  rpcSettingsStore: {
    rpcSettings: { enabled: true, interface: 'Baz', port: 'Qux' }
  }
};

test('should render correctly', () => {
  const component = shallowWithIntl(<Network {...props} />);

  expect(shallowToJson(component)).toMatchSnapshot();
});

test('should render correctly when not fully loaded', () => {
  const component = shallowWithIntl(
    <Network
      {...props}
      rpcSettingsStore={{ rpcSettings: { enabled: false } }}
      netPeersStore={{ netPeers: { realPeers: [] } }}
    />
  );

  expect(shallowToJson(component)).toMatchSnapshot();
});
