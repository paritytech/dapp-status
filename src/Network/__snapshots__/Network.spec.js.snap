// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Section
  title={
    <FormattedMessage
      defaultMessage="Network Settings"
      id="dapp.status.network.title"
      values={Object {}}
    />
  }
>
  <Form
    as="form"
  >
    <InjectIntl(Field)
      label={
        <FormattedMessage
          defaultMessage="Chain"
          id="dapp.status.network.chainLabel"
          values={Object {}}
        />
      }
      readOnly={true}
      value="Foo"
    />
    <FormGroup
      widths={2}
    >
      <InjectIntl(Field)
        label={
          <FormattedMessage
            defaultMessage="RPC Enabled"
            id="dapp.status.network.rpcEnabled"
            values={Object {}}
          />
        }
        readOnly={true}
        value="Yes"
      />
      <InjectIntl(Field)
        label={
          <FormattedMessage
            defaultMessage="Network Port"
            id="dapp.status.network.netPort"
            values={Object {}}
          />
        }
        readOnly={true}
        value={4}
      />
    </FormGroup>
    <FormGroup
      widths={2}
    >
      <InjectIntl(Field)
        label={
          <FormattedMessage
            defaultMessage="RPC Interface"
            id="dapp.status.network.rpcInterface"
            values={Object {}}
          />
        }
        readOnly={true}
        value="Baz"
      />
      <InjectIntl(Field)
        label={
          <FormattedMessage
            defaultMessage="RPC Port"
            id="dapp.status.network.rpcPort"
            values={Object {}}
          />
        }
        readOnly={true}
        value="Qux"
      />
    </FormGroup>
    <InjectIntl(Field)
      label={
        <FormattedMessage
          defaultMessage="Enode"
          id="dapp.status.network.enode"
          values={Object {}}
        />
      }
      readOnly={true}
      showCopyButton={true}
      value="Bar"
    />
  </Form>
  <LowerCaseStatistic
    size="tiny"
    widths={1}
  >
    <Statistic>
      <StatisticValue>
        1/2/3
      </StatisticValue>
      <StatisticLabel>
        <FormattedMessage
          defaultMessage="Active/Connected/Max Peers"
          id="dapp.status.mining.netPeersLabel"
          values={Object {}}
        />
      </StatisticLabel>
    </Statistic>
  </LowerCaseStatistic>
</Section>
`;

exports[`should render correctly when not fully loaded 1`] = `
<Section
  title={
    <FormattedMessage
      defaultMessage="Network Settings"
      id="dapp.status.network.title"
      values={Object {}}
    />
  }
>
  <Form
    as="form"
  >
    <InjectIntl(Field)
      label={
        <FormattedMessage
          defaultMessage="Chain"
          id="dapp.status.network.chainLabel"
          values={Object {}}
        />
      }
      readOnly={true}
      value="Foo"
    />
    <FormGroup
      widths={2}
    >
      <InjectIntl(Field)
        label={
          <FormattedMessage
            defaultMessage="RPC Enabled"
            id="dapp.status.network.rpcEnabled"
            values={Object {}}
          />
        }
        readOnly={true}
        value="No"
      />
      <InjectIntl(Field)
        label={
          <FormattedMessage
            defaultMessage="Network Port"
            id="dapp.status.network.netPort"
            values={Object {}}
          />
        }
        readOnly={true}
        value={4}
      />
    </FormGroup>
    <FormGroup
      widths={2}
    >
      <InjectIntl(Field)
        label={
          <FormattedMessage
            defaultMessage="RPC Interface"
            id="dapp.status.network.rpcInterface"
            values={Object {}}
          />
        }
        readOnly={true}
      />
      <InjectIntl(Field)
        label={
          <FormattedMessage
            defaultMessage="RPC Port"
            id="dapp.status.network.rpcPort"
            values={Object {}}
          />
        }
        readOnly={true}
      />
    </FormGroup>
    <InjectIntl(Field)
      label={
        <FormattedMessage
          defaultMessage="Enode"
          id="dapp.status.network.enode"
          values={Object {}}
        />
      }
      readOnly={true}
      showCopyButton={true}
      value="Bar"
    />
  </Form>
  <LowerCaseStatistic
    size="tiny"
    widths={1}
  >
    <Statistic>
      <StatisticValue>
        0/0/0
      </StatisticValue>
      <StatisticLabel>
        <FormattedMessage
          defaultMessage="Active/Connected/Max Peers"
          id="dapp.status.mining.netPeersLabel"
          values={Object {}}
        />
      </StatisticLabel>
    </Statistic>
  </LowerCaseStatistic>
</Section>
`;
